### DISCUSSION TITLE
Podman 4 can't run image from repository due to missed libtinfo.so

### DISCUSSION THREAD

**Discussion by PavelSosin-320** [containers/podman#14070] [General]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

kind bug

/kind feature

**Description**
Podman fails to execute run command due to missed dependency libtinfo with error:
error while loading shared libraries: libtinfo.so.6: cannot open shared object file: No such file or directory

**Steps to reproduce the issue:**

1. execute podman run command providing image spec as parameter with log-level=debug option
2.

3.

**Describe the results you received:**
Error message:
error while loading shared libraries: libtinfo.so.6: cannot open shared object file: No such file or directory
**Describe the results you expected:**
Open the panel for repository selection

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Client:       Podman Engine
Version:      4.0.3
API Version:  4.0.3
Go Version:   go1.18
Built:        Fri Apr  1 21:21:54 2022
OS/Arch:      linux/amd64

```

**Output of `podman info --debug`:**

```
I suppose that it is not so relevant
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-4.0.3-1.fc36.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Fedora 36 Workstation modular
Indeed, libtinfo.so.6 is not installed and I can't find the package that provides it. ncurses 6.2-9 is installed.
It looks like it is not provided with many Workstation versions according to Google and ncurses for Gnome 4 functionality is messy.
Could you provide the image repository defaulting as a configuration to make solution more convenient?

**Comment by giuseppe** [comment]
how have you installed podman?

Is `ncurses-libs` installed?

**Reply by PavelSosin-320** [reply]
@giuseppe ncurses-libs is installed. Yes, I tried to install / reinstall everything that related to curses because invert dependencies search said me that libtinfo.so must be brought as a part of courses. It is still missed. I found using Google search that libtinfo is missed not only in the case of podman on  Fedora WS, it is missed forever. Maybe, it is wrong to expect that curses is a part of Gnome 4.2 GUI stack because courses simulate GUI in the text mode and requires gnome-terminal that simulates text terminal on the Gnome GUI. The current shape of Fedora workstation GUI is so good that text mode is completely redundant except scripting, systemd-unit-like files and  maintenance cases. Very few users will cry for curses. Fedora 36 market penetration looks even better than still not enough polished Windows 11 according to my Win 11 insider version.

**Comment by rhatdan** [comment]
This looks like an SELinux labeling issue.

sudo dnf -y reinstall container-selinux

See if this fails.  If it does then  you have an issue with flatpack-selinux snapd-selinux ... being installed 

https://bugzilla.redhat.com/show_bug.cgi?id=2056303

**Comment by PavelSosin-320** [comment]
@giuseppe This is the enigma: ncurses, ncurses-libs ver 6.2-9 (the recent version) are installed but don't provide libtinfo.so. And I can't find the  package that provides it. I searched in Google and found many reference to the same issue in Ubuntu etc. libtinfo is listed as the ncurses dependency but doesn't come. I afraid that ncurses itself or its installation is broken for the Gnome 4 / Wayland backend. Maybe ncursesw is more suitable for Gnome-terminal?

**Comment by PavelSosin-320** [comment]
@rhatdan OMG Yes, I had issues with flatpack-selinux, etc. like other people that upgraded from Fedora 35 to 36. Now 36 GA release looks frozen due to this issue because many people that upgraded from Fedora 35 to 36 recently. Zdenek is working on it. But my specific problem is that libtinfo doesn't exist, really doesn't exist for Fedora 36 or exists somewhere in the broken repository. Maybe, some repositories are IPV4 only or curses has gone in Gnome's 4.2 terminal. Indeed, today ncurses provides only:
The libraries are:

	libncurses.a       (normal)
	libncurses.so      (shared)
	libncurses_g.a     (debug and trace code enabled)
	libncurses_p.a     (profiling enabled)

	libpanel.a         (normal)
	libpanel.so        (shared)
	libpanel_g.a       (debug and trace code enabled)

	libmenu.a          (normal)
	libmenu.so         (shared)
	libmenu_g.a        (debug enabled)

	libform.a          (normal)
	libform.so         (shared)
	libform_g.a        (debug enabled)
libtinfo has to come from another package or replaced with something else ncursesw ? 
P.S. It's very hard to find any product that has libtinfo dependency and doesn't fail For example: [purescript failure on libtinfo  dependency](https://github.com/purescript/purescript/issues/4253). The terminal info is nonsense for Wayland / Gnome terminal - it is always x-terminal as the process environment tells me.

**Comment by rhatdan** [comment]
Could you put the machine into permissive mode and see if the problem goes away?

**Comment by PavelSosin-320** [comment]
@vrothberg Could you speed up implementation of your proposal described in [managing container registries](https://www.redhat.com/sysadmin/manage-container-registries) to get rid the use of curses graphics and pass registry's short name as an option? Taking into account endless number of combinations DesktopManager TerminalSimulator CursesImplementation can produce endless stream of issues  like "missed dependency", "mouse event is not processed", "ugly graphics", etc. 
Interaction with curses graphics also is not good for scripting.

**Comment by vrothberg** [comment]
I do not understand the request.  To me it looks like your installation is broken.

**Comment by giuseppe** [comment]
let's convert to a discussion since there is no issue in Podman itself
